source/routing/raptor_api.cpp

```diff
@@ -1365,6 +1368,75 @@ void filter_late_journeys(RAPTOR::Journeys& journeys, const NightBusFilter::Para
     }
 }
 
+bool find_stop_area_of_interest_through_journey(const navitia::routing::Journey& journey,
+                                                const std::string& stop_area_uri,
+                                                const navitia::type::VehicleJourney* vj_to_skip) {
+    bool found = false;
+
+    for (const auto& section : journey.sections) {
+        auto order = section.get_in_st->order();
+        bool reach_end = false;
+        for (const auto* vj = section.get_in_st->vehicle_journey; vj;
```

Comment:
```suggestion
        // because of stay-ins, we may have several vj in one section
        for (const auto* vj = section.get_in_st->vehicle_journey; vj;
```